libfs/fs.go

```diff
@@ -108,31 +109,60 @@ func (e ErrNotADirectory) Error() string {
 	return fmt.Sprintf("%s is not a directory", e.Name)
 }
 
+type accessType int
+
+const (
+	readwrite accessType = iota
```

Comment:
nit: capitalization here confuses me.
also, maybe add a comment on `readwriteNoCreate` since it's non-obvious to me what create means without looking at the context of the PR.

Reply:
Re: capitalization: yeah, I was trying to be consistent with the `Readonly` capitalization we have (which maybe we inherited from somewhere else?  I don't remember.)  Anyway, I can change it if people feel strongly about it.

Will add a comment.